test_ansible_shell.yml

```diff
@@ -0,0 +1,16 @@
+---
+- name: Verify that ansible can correctly pass arguments to ansible in shell
+  hosts: all
+  gather_facts: False
+  tasks:
+    - name: Write inventory file
+      blockinfile:
+        path: '{{ playbook_dir }}/inventory.ansible.shell'
+        block: |
+          localhost ansible_connection=local
+        state: present
+        create: True
+
+    - name: Run Playbook
+      shell: "ANSIBLE_NOCOLOR=1 ansible-playbook -i './inventory.ansible.shell' ./ping.yml"
```

Comment:
last question, does this fail if return code is non-zero?

Reply:
Yep, shell tasks are always marked failed if they return a non-zero, unless you supply `failed_when` conditions